pkg/template/processor.go

```diff
@@ -0,0 +1,111 @@
+package template
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/go-logr/logr"
+	templatev1 "github.com/openshift/api/template/v1"
+	"github.com/openshift/library-go/pkg/template/generator"
+	"github.com/openshift/library-go/pkg/template/templateprocessing"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/api/errors"
+	errs "github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"context"
+)
+
+type Processor struct {
+	cl     client.Client
+	scheme *runtime.Scheme
+	log    logr.Logger
+}
+
+func NewProcessor(cl client.Client, scheme *runtime.Scheme, log logr.Logger) *Processor {
+	return &Processor{cl: cl, scheme: scheme, log: log}
+}
+
+func (p Processor) ProcessAndApply(tmplContent []byte, values map[string]string) error {
+	objs, err := p.Process(tmplContent, values)
+	if err != nil {
+		return err
+	}
+
+	return apply(p.cl, objs, p.log)
+}
+
+func apply(cl client.Client, objs []runtime.RawExtension, rq logr.Logger) error {
+	for _, obj := range objs {
```

Comment:
> I would prefer, to sort all template objects as per our need outside of this code.

sorting out of this particular code for sure. I would do it after the process logic when you get the slice of objects.  

> it has to make sure the template is in correct order for addition or deletion(reverse order)

You can take a look at tenant service - there is done exactly this 

> otherwise while getting the template from host should be in correct order of dependency.

I wouldn't really rely on this. I would expect that the template that is retrieved from host can be in a wrong order

Reply:
> for namespace creation verification, it makes sense to do it upfront. I will add it.

please, check the tenant service

Reply:
@dipak-pawar it would be great if you first check if this dependent resource creation is an issue for what usecase/scenario.  Can you please verify this https://jira.coreos.com/browse/CRT-197?focusedCommentId=119466&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-119466

Reply:
@nurali-techie thanks for reminding me, I have already considered this, we'll be adding sorting logic for this